pkg/kapp/clusterapply/cluster_change.go

```diff
@@ -146,7 +146,7 @@ func (c *ClusterChange) WaitOp() ClusterChangeWaitOp {
 		// TODO associated resources
 		// If existing resource is not in a "done successful" state,
 		// indicate that this will be something we need to wait for
-		existingResState, _, existingErr := c.convergedResFactory.New(c.change.ExistingResource(), nil).IsDoneApplying()
+		existingResState, _, existingErr := c.convergedResFactory.New(c.change.ClusterOriginalResource(), nil).IsDoneApplying()
```

Comment:
So we are using ClusterOriginalResource here so that we have access to status?
(I believe that is the whole point but just clarifying)

Reply:
Yep, that's correct.

Reply:
Makes sense!

Reply:
nit: change `existingResState` to `resState` or `originalResState` and `existingErr` to `err`. 

Considering that `Change` has existingRes and ClusterOriginalRes, we shouldn't mix the variable name to avoid confusion.